cms/models.py

```diff
@@ -1,18 +1,53 @@
 """
 Page models for the CMS
 """
-import json
-
 from django.db import models
+from django.http.response import Http404
 from django.utils.text import slugify
 from wagtail.admin.edit_handlers import FieldPanel, StreamFieldPanel
 from wagtail.core.fields import RichTextField, StreamField
-from wagtail.core.models import Page
-from wagtail.images.edit_handlers import ImageChooserPanel
 from wagtail.images.models import Image
+from wagtail.images.edit_handlers import ImageChooserPanel
+from wagtail.core.models import Page
+from wagtail.core.blocks import StreamBlock
+
+from cms.blocks import ResourceBlock, LearningAreaBlock
+
+
+class BootcampProductChildPage(Page):
+    """
+    Abstract page representing a child of Bootcamp Product Page
+    """
+
+    class Meta:
+        abstract = True
+
+    subpage_types = []
+    parent_page_types = ["BootcampRunPage"]
+
+    # disable promote panels, no need for slug entry, it will be autogenerated
+    promote_panels = []
+
+    @classmethod
+    def can_create_at(cls, parent):
+        # You can only create one of these page under bootcamp product.
+        return (
+            super().can_create_at(parent)
+            and parent.get_children().type(cls).count() == 0
+        )
+
+    def save(self, *args, **kwargs):
+        # autogenerate a unique slug so we don't hit a ValidationError
+        if not self.title:
+            self.title = self.__class__._meta.verbose_name.title()
+        super().save(*args, **kwargs)
```

Comment:
Don't we need that line before `super().save()` like we did in xPRO?
```self.slug = slugify("{}-{}".format(self.get_parent().id, self.title))```

Reply:
@umarmughal824 is correct here. This will always generate the same title so the same kind of child page which in turn will translate into the same slug, thereby creating an integrity error at save.

Reply:
not generating the same title, I believe, parent_id would be the same in child.

Reply:
I don't see `parent_id` anywhere 🤷‍♂️

Reply:
My question is, would be the parent_id be different or the same in that case ?

```
self.slug = slugify("{}-{}".format(self.get_parent().id, self.title))
```